test/Sema/option-set-empty.swift

```diff
@@ -0,0 +1,6 @@
+// RUN: %target-typecheck-verify-swift
+
+struct MyOptions: OptionSet {
+    var rawValue: Int
+    static let none = MyOptions(rawValue: 0) // expected-warning {{static property 'none' produces an empty option set}} expected-note {{use [] to silence this warning}}{{32-45=([])}}
```

Comment:
Can you add some test cases that *shouldn't* cause the error? I can think of a few:

- Other types
- Other constructors of MyOptions
- Values other than 0
- `var`s

Reply:
Non-static `let` too, maybe! And anything else you can think of.

Reply:
Still would like you to add this last test too. :-)

Reply:
Sorry! I did, but then rolled it back to try to figure out where it was crashing. I think I figured it out now though. Just need to confirm it.

Reply:
I added the tests back but I had to make the non-static let of another type or else I'd get a build error because a struct cannot contain itself. Should I add that case anyways and expect the error?

Reply:
Ah, right. And if it's for another type it's not a new test. I guess it's still good to add the test case and expect the one error to make sure the second one doesn't pop up.